### PR TITLE
Adding few more function to ADHybridHealth swagger spec

### PR DESCRIPTION
**Pull request by gamitra**
This Pull Request contains the Swagger specification for Microsoft.ADHybridHealthService RP.
This RP is for Azure Active Directory Connect Health service, which has been GA-ed for last 2 years.
Azure Active Directory (Azure AD) Connect Health helps you monitor and gain insights into your on-premises identity infrastructure and the synchronization services. It enables you to maintain a reliable connection to Office 365 and Microsoft Online Services by providing monitoring capabilities for your key identity components such as Active Directory Federation Services (AD FS) servers, Azure AD Connect servers (also known as Sync Engine), Active Directory domain controllers, etc. It also makes the key data points about these components easily accessible so that you can get usage and other important insights to make informed decisions.
The information is presented in the Azure AD Connect Health portal. In the Azure AD Connect Health portal, you can view alerts, performance monitoring, usage analytics, and other information. Azure AD Connect Health enables the single lens of health for your key identity components in one place.
More details about this RP can be found at:
https://docs.microsoft.com/en-us/azure/active-directory/connect-health/active-directory-aadconnect-health

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by dsgouda** [comment]
@gamitra will review soon

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json [hunk 1]
```diff
@@ -478,6 +478,122 @@
...
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": " ",
```
Comment (dsgouda) [inline specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: Please provide a meaningful description.

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json [hunk 2]
```diff
@@ -478,6 +478,122 @@
...
+    "/providers/Microsoft.ADHybridHealthService/addsservices/{serviceName}/features/{featureName}/userpreference": {
+      "get": {
+        "tags": [ "Adds" ],
+        "description": "Gets the user preferences for a given feature.",
+        "x-ms-examples": {
+          "addsServices_getUserPreference": {
+            "$ref": "./examples/UserPreference.json"
+          }
+        },
+        "operationId": "addsServices_getUserPreference",
```
Comment (dsgouda) [inline specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: A recommended operation id would be `UserPreferences_Get` or `AddsServicesUserPreference_Get`
Also, just out of curiosity, what does `Adds` stand for

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json [hunk 3]
```diff
@@ -478,6 +478,122 @@
...
+            "description": "The name of the feature.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (dsgouda) [inline specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: Please provide meaningful description for all operations
Please confirm whether this is a long running operation

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json [hunk 4]
```diff
@@ -759,22 +944,285 @@
...
+          },
+          {
+            "name": "$filter",
+            "in": "query",
+            "description": "The server property filter to apply.",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "name": "withDetails",
```
Comment (dsgouda) [inline specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: A better name for this parameter would be `showDetails` or `fetchDetails`

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json [hunk 5]
```diff
@@ -510,6 +626,75 @@
...
+            "required": false,
+            "type": "string",
+            "format": "date-time"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (dsgouda) [inline specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: Please add `x-ms-pageable` extension

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json [hunk 6]
```diff
@@ -759,22 +944,285 @@
...
+          "nextLinkName": null
+        }
+      }
+    },
+    "/providers/Microsoft.ADHybridHealthService/addsservices/{serviceName}/replicationstatus": {
+      "get": {
+        "tags": [ "Adds" ],
+        "description": "Gets Replication status for a given Active Directory Domain Service, that is onboarded to Azure Active Directory Connect Health.",
+        "x-ms-examples": {
+          "addsServices_getReplicationStatus": {
```
Comment (dsgouda) [inline specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: Please rename operationid to `ReplicationStatus_Get` or `AddsServicesReplicationStatus_Get`

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json [hunk 7]
```diff
@@ -759,22 +944,285 @@
...
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "nextPartitionKey",
+            "in": "query",
+            "description": "The next partition key to query for.",
+            "required": true,
+            "type": "string",
+            "enum": [ " " ],
```
Comment (dsgouda) [inline specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: Please create a model definition and refer its schema for enum parameters

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json [hunk 8]
```diff
@@ -1671,6 +2157,75 @@
...
+    "/providers/Microsoft.ADHybridHealthService/services/{serviceName}/metrics/{metricName}/groups/{groupName}": {
+      "get": {
+        "tags": [ "Metrics" ],
+        "description": "Gets the server related metrics for a given metric and group combination.",
+        "x-ms-examples": {
+          "service_listMetrics": {
+            "$ref": "./examples/Service_MetricSets.json"
+          }
+        },
+        "operationId": "service_listMetrics",
```
Comment (dsgouda) [inline specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: Similar to all other operationids, please rename this to `ServiceMetrics_List` and `Metrics_List`

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json [hunk 9]
```diff
@@ -5061,6 +5616,27 @@
         }
       }
     },
+    "ReplicationDetailsList": {
+      "description": "The list of replication details.",
+      "type": "object",
+      "properties": {
```
Comment (dsgouda) [inline specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: Please mark properties `readOnly` wherever applicable.

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please address warnings in the CI linter build

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json (hunk 2 above)
Comment (gamitra) [reply specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: @dsgouda, ok I will use AddsServicesUserPreference_get, since it aligns with the naming convention I have used for the previous operations also.
Adds is Active Directory Domain Service,

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json (hunk 3 above)
Comment (gamitra) [reply specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: @dsgouda, did not get your point. What is a meaningful description for an operation which does not display any result? Do you have an example for me to follow?

Comment (gamitra) [reply specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: @dsgouda , also will be good if you can help me with an example for long running operation.

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json (hunk 4 above)
Comment (gamitra) [reply specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: This APIs are already in production and being used by UI, so I cant rename any parameter.

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json (hunk 3 above)
Comment (dsgouda) [reply specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: Here is an example. 
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/network/resource-manager/Microsoft.Network/stable/2018-05-01/applicationGateway.json#L72

Comment (gamitra) [reply specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: So none of the operations listed in my swagger spec are long running.
Do I still need to add that flag? All the delete, add,get are targetted operations with a max latency of 6s.

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json (hunk 2 above)
Comment (gamitra) [reply specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: This is done.

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json (hunk 1 above)
Comment (gamitra) [reply specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: Added description text for all operations.

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json (hunk 3 above)
Comment (gamitra) [reply specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: This one I left out, waiting for the clarification for my previous qs.

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json (hunk 5 above)
Comment (gamitra) [reply specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: Done.

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json (hunk 4 above)
Comment (gamitra) [reply specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: Left this out also, because this is a production code.

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json (hunk 6 above)
Comment (gamitra) [reply specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: Done.

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json (hunk 7 above)
Comment (gamitra) [reply specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: Left this out, as mentioned above.

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json (hunk 8 above)
Comment (gamitra) [reply specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: Done.

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json (hunk 9 above)
Comment (gamitra) [reply specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: Need some help understanding this comment. I left it out of the current iteration. Once I get more clarification, I will add this.

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json (hunk 3 above)
Comment (dsgouda) [reply specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: Given the latency you do not need to set the flag.

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json (hunk 9 above)
Comment (dsgouda) [reply specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: If the model properties must have public getters, `readOnly` flag is necessary. If not, it must be set for each of the properties of the model. This ensures better quality SDK

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json (hunk 7 above)
Comment (dsgouda) [reply specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: The way the enum is modeled here is not the issue. We discourage usage of inline definitions of schemas (which is enum here). Please follow an example [here](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/analysisservices/resource-manager/Microsoft.AnalysisServices/stable/2017-08-01/analysisservices.json#L99) and move the schema definition into definitions section

**Comment by dsgouda** [comment]
@gamitra Please address linter errors/warnings [here](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/389463748)

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json (hunk 7 above)
Comment (gamitra) [reply specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: Got it now. Will send another iteration with the change.

**Comment by gamitra** [comment]
@dsgouda, working on the linter error. The build looks flaky, most of the builds are failing with connection timeout error. 
I will update you once I have a good build ready for your review.

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json (hunk 9 above)
Comment (gamitra) [reply specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: Thanks for explaining. All the model properties for our service are public getters.

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json (hunk 5 above)
Comment (gamitra) [reply specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: I revisited this operation,. its not a list. It's a get function. I will change the name of the operation to avoid confusion.

File: specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json (hunk 7 above)
Comment (gamitra) [reply specification/adhybridhealthservice/resource-manager/Microsoft.ADHybridHealthService/stable/2014-01-01/ADHybridHealthService.json]: @dsgouda , I made the changes for the enum definition as suggested by you. But after that build is failing:
https://travis-ci.org/Azure/azure-rest-api-specs/jobs/389568900

Can you please help me understand what am I doing wrong here? How can I correct that? Thanks for helping.

**Comment by gamitra** [comment]
@dsgouda, this started failing after I tried to move the enum to definition section as suggested by you.
However, I am unable to figure out why is this failure, been trying to resolve it from yesterday. Do you see any obvious issue?
It would be really helpful, if you can point me to the issue.

**Comment by dsgouda** [comment]
@gamitra looked into the CI failures and I think I know the issue. Looks like we can't have schema definitions for `query` parameters.
tldr;
Please revert to the earlier inline definition for the enum and that should fix the CI issues.
Lastly, could you explain why you need an empty enum here.
@fearthecowboy FYI

**Comment by dsgouda** [comment]
@gamitra please check the Json for correctness

**Review by dsgouda — APPROVED** [review]
LGTM subject to CIs passing

**Comment by gamitra** [comment]
@dsgouda , making the CIs work.. give me some time, I will let you know when the PR is ready for you to take a look.
I am submitting iteration and  trying to make the CIs pass.

**Comment by gamitra** [comment]
@dsgouda , Build passed. I see you have signed off also. Thank You for that.
How do I merge the pull request? The "Merge pull request" button for me is showing disabled..
Last review, the reviewer merged it for me. 
Can you please help me merge this PR?

**Comment by dsgouda** [comment]
Done.

**Comment by gamitra** [comment]
Thank You so much for your help. Really appreciate it!
